main.js

```diff
@@ -0,0 +1,149 @@
+var VideoModel = Backbone.Model.extend({
+  defaults: function () {
+    return {
+      videoId: "",
+      videoThumbnamil: "",
+      videoTitle: "",
+      videoDesc: "",
+    };
+  },
+});
+
+// Videos collection - add API fetch here
+var VideosCollection = Backbone.Collection.extend({ 
+  url: function (topic) {
```

Comment:
You don't need this `topic` parameter as you're never calling `url` as a function.

You've half-way implemented two ways to search for videos. You need to pick one way and finish it.